### PR TITLE
HDDS-5473. Bump jetty version to 9.4.43.v20210629

### PR DESCRIPTION
**Pull request by elek**
JIRA: https://issues.apache.org/jira/browse/HDDS-5473

## What changes were proposed in this pull request?

Recent jetty versions contain multiple security [vulnerabilities](https://github.com/eclipse/jetty.project/security). Most of them are low priorities, but it seems to be better to update the jetty versions. 

## How was this patch tested?

Full CI passed on my fork + checked if the `share/ozone/lib` folder contains only jetty jars with the new version.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -96,7 +96,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
 
     <failIfNoTests>false</failIfNoTests>
     <maven.test.redirectTestOutputToFile>true</maven.test.redirectTestOutputToFile>
-    <jetty.version>9.4.35.v20201120</jetty.version>
+    <jetty.version>9.4.41.v20210516</jetty.version>
```
Comment (ayushtkn) [inline pom.xml:99]: Any reason for not using ``9.4.43.v20210629``? Or we just want to cover the CVE fix?

**Comment by jojochuang** [comment]
triggering the test again. i think it's fine to use 9.4.41 to cover known issues.

File: pom.xml (hunk 1 above)
Comment (adoroszlai) [reply pom.xml:99]: We may want to go with 9.4.43, as the most recent CVE is only fixed in that version: https://github.com/eclipse/jetty.project/security/advisories/GHSA-vjv5-gp2w-65vm

**Comment by adoroszlai** [comment]
@elek Thanks for the fix.  I've updated the PR with the latest Jetty version to cover most recent CVE, too.  I've also improved `update-jar-report.sh` to remove Jetty versions (which do not fit standard version number scheme).  Please let me know what you think.

**Review by ayushtkn — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @elek for the fix, @ayushtkn and @jojochuang for the review.

**Pull request merged**